ansible/build-node.yml

```diff
@@ -2,29 +2,31 @@
 - hosts: 127.0.0.1
   connection: local
 
-  vars:
-    - cm_buildimage: "caremonkey/cm-builder"
-    - ecr_image_name: "984601232468.dkr.ecr.us-west-2.amazonaws.com/{{ cm_buildimage }}:latest"
-
   tasks:
-    - name: Get Login command for private AWS ECR registry
-      command: aws ecr get-login --region us-west-2
-      register: docker_login_command
+    - name: Copy ECR login shell script
+      become: yes
+      copy: src=files/ecr-login.sh dest=/usr/local/bin/ecr-login.sh mode=0755
+
+    - name: Run the ECR login now for good measure
+      become: yes
+      become_user: buildkite-agent
+      command: /usr/local/bin/ecr-login.sh
 
-    - name: Login to the registry
+    - name: Add it to the crontab
       become: yes
-      command: "{{ docker_login_command.stdout_lines.0 }}"
+      become_user: buildkite-agent
+      cron:
+        name: "ECR login"
+        job: "/usr/local/bin/ecr-login.sh 2>&1 /dev/null"
+        state: present
+        minute: "*/10"
 
     - name: Pull supporting Docker images
       become: yes
-      docker_image:
-       name: "{{ item }}"
+      become_user: buildkite-agent
+      command: docker pull "{{ item }}"
       with_items:
+        - "984601232468.dkr.ecr.us-west-2.amazonaws.com/caremonkey/cm-builder:3"
```

Comment:
Meant to be 3. The diff between this and v2 is small, and v2 will get pulled automatically. This way we bake the AMI machine image with 3 already pre-loaded for later on ;)